homeassistant/components/ws66i/media_player.py

```diff
@@ -0,0 +1,199 @@
+"""Support for interfacing with WS66i 6 zone home audio controller."""
+from datetime import timedelta
+import logging
+
+from homeassistant import core
+from homeassistant.components.media_player import MediaPlayerEntity
+from homeassistant.components.media_player.const import (
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.const import CONF_IP_ADDRESS, STATE_OFF, STATE_ON
+from homeassistant.helpers import entity_platform
+
+from .const import CONF_SOURCES, DOMAIN, SERVICE_RESTORE, SERVICE_SNAPSHOT, WS66I_OBJECT
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+SUPPORT_WS66I = (
+    SUPPORT_VOLUME_MUTE
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_SELECT_SOURCE
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+@core.callback
+def _get_sources_from_dict(data):
+    sources_config = data[CONF_SOURCES]
+
+    # Dict index to custom name
+    source_id_name = {int(index): name for index, name in sources_config.items()}
+
+    # Dict custom name to index
+    source_name_id = {v: k for k, v in source_id_name.items()}
+
+    # List of custom names
+    source_names = sorted(source_name_id.keys(), key=lambda v: source_name_id[v])
+
+    return [source_id_name, source_name_id, source_names]
```

Comment:
This would be much cleaner if you used a dataclass 

https://docs.python.org/3/library/dataclasses.html

Reply:
How would that look like? Would I be returning a class object from this function? I would have to initialize a dataclass and then access those fields using "dataclasses.fields()"